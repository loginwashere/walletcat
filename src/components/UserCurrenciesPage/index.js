import React, { PropTypes } from 'react'
import { ListGroup } from 'react-bootstrap'
import AppCurrency from '../AppCurrency'

const UserCurrenciesPage = ({ userCurrencyIds, currencies, userCurrencies, dispatch }) => (
  <ListGroup>
    {userCurrencyIds.map(id => {
      const userCurrency = userCurrencies[id]
      const currency = userCurrency && currencies[userCurrency.currencyId]
      return (
        currency &&
        userCurrency &&
        <AppCurrency key={currency.id}
                     currency={currency}
                     userCurrency={userCurrency}
                     dispatch={dispatch} />
      )
    })}
  </ListGroup>
)

UserCurrenciesPage.propTypes = {
  userCurrencyIds: PropTypes.array.isRequired,
  currencies: PropTypes.object.isRequired,
  userCurrencies: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired
}

export default UserCurrenciesPage
