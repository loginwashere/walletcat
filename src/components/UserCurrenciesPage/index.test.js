import React from 'react'
import renderer from 'react-test-renderer'
import {
  userCurrencySeeder,
  currencySeeder
} from '../../../server/seeds'
import UserCurrenciesPage from '.'

describe('components:Login:', () => {
  it('renders without crashing and without required data', () => {
    const dispatch = jest.fn()
    const userCurrency = userCurrencySeeder.items[0]
    const currency = currencySeeder.items[0]
    const currencies = { [currency.id]: currency }
    const userCurrencies = { [userCurrency.id]: userCurrency }
    const tree = renderer.create(
        <UserCurrenciesPage dispatch={dispatch}
                            userCurrencies={userCurrencies}
                            currencies={currencies} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without crashing and with required data', () => {
    const dispatch = jest.fn()
    const userCurrency = userCurrencySeeder.items[0]
    const currency = currencySeeder.items[0]
    const currencies = { [currency.id]: currency }
    const userCurrencies = { [userCurrency.id]: userCurrency }
    const userCurrencyIds = [userCurrency.id]
    const tree = renderer.create(
        <UserCurrenciesPage dispatch={dispatch}
                            userCurrencyIds={userCurrencyIds}
                            userCurrencies={userCurrencies}
                            currencies={currencies} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})