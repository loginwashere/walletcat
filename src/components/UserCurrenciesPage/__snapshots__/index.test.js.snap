exports[`components:Login: renders without crashing and with required data 1`] = `
<div
  className="list-group">
  <li
    className="list-group-item">
    <button
      className="pull-right btn btn-xs btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      Remove from My Currencies
    </button>
    <h4
      className="list-group-item-heading">
      AED
    </h4>
    <p
      className="list-group-item-text">
      United Arab Emirates Dirham
    </p>
  </li>
</div>
`;

exports[`components:Login: renders without crashing and without required data 1`] = `
<ul
  className="list-group" />
`;
